static/app/components/organizations/multipleEnvironmentSelector.tsx

```diff
@@ -67,40 +61,21 @@ type State = {
  * Note we only fetch environments when this component is mounted
  */
 class MultipleEnvironmentSelector extends React.PureComponent<Props, State> {
-  static defaultProps: DefaultProps = {
-    value: [],
-  };
-
   state: State = {
     selectedEnvs: new Set(this.props.value),
-    hasChanges: false,
   };
 
   componentDidUpdate(prevProps: Props) {
     // Need to sync state
     if (this.props.value !== prevProps.value) {
-      this.syncSelectedStateFromProps();
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({selectedEnvs: new Set(this.props.value)});
     }
   }
 
-  syncSelectedStateFromProps = () =>
-    this.setState({selectedEnvs: new Set(this.props.value)});
-
-  /**
-   * If value in state is different than value from props, propagate changes
-   */
-  doChange = (environments: string[]) => {
-    this.props.onChange(environments);
-  };
-
-  /**
-   * Checks if "onUpdate" is callable. Only calls if there are changes
-   * @param selectedEnvs optional parameter passed to onUpdate representing
-   * an array containing a directly selected environment (not multi-selected)
-   */
-  doUpdate = (selectedEnvs?: string[]) => {
-    this.setState({hasChanges: false}, () => this.props.onUpdate(selectedEnvs));
-  };
+  get hasChanges() {
```

Comment:
didn't need the `hasChanges` state.. we can just compute it.

Succeedes #32379